### ISSUE TITLE
Outlier removal breaks in gwdetchar-lasso-correlation

### ISSUE THREAD

**Issue by macedo22** [gwdetchar/gwdetchar#154]
I was just testing the new `gwpy-0.12.2` and `gwdetchar` installs at LLO using `detchar`'s environment, but ran into an issue. `gwdetchar-lasso-correlation` now breaks on the outlier removal in the beginning, but that function hasn't been changed. My reason for thinking it might just be me is because in the merged PR https://github.com/gwdetchar/gwdetchar/pull/134 , @duncanmmacleod  used outlier removal (`-O 2.5`) in his command line when he was testing the lasso updates, and he provided examples of pages he generated - so I assume he didn't encounter this issue. Here is an example error message below. It seems like it's reporting an indexing or transposition logic error:

```
(gwdetchar-dev) [alexandra.macedo@ldas-pcdev5 gwdetchar]$ gwdetchar-lasso-correlation 1171432800 1171453380 -O 2 -o ~/public_html/detchar/LASSO/test/lasso-update -i L1 -a 0.1 -t .0001 -c .95 -J 16 -j 16 -f ~/public_html/detchar/channel-lists/chans.txt
-- Loading primary channel data
Attempting to access data from frames...
Determined best frametype as 'SenseMonitor_hoft_L1_M'
Reading SenseMonitor_hoft_L1_M frames: |██████████████| 6/6 (100%) ETA 00:00
-- Removing outliers above 2.000000 sigma
-- Found 2 outliers in L1:DMT-SNSH_EFFECTIVE_RANGE_MPC.mean, recursively removing
Completed 1 removal cycles
13 outliers remain
Completed 2 removal cycles
6 outliers remain
Completed 3 removal cycles
5 outliers remain
Traceback (most recent call last):
  File "/home/alexandra.macedo/opt/gwdetchar-dev/bin/gwdetchar-lasso-correlation", line 274, in <module>
    remove_outliers(primaryts, args.remove_outliers)
  File "/home/alexandra.macedo/opt/gwdetchar-dev/bin/gwdetchar-lasso-correlation", line 107, in remove_outliers
    ts[:] = (spline(ts[outliers].times.value) * unit)
  File "/usr/lib64/python2.7/site-packages/astropy/units/quantity.py", line 1049, in __setslice__
    self.view(np.ndarray).__setslice__(i, j, self._to_own_unit(value))
ValueError: could not broadcast input array from shape (5) into shape (343)
(gwdetchar-dev) [alexandra.macedo@ldas-pcdev5 gwdetchar]$
```

@duncanmmacleod let me know if you ran into anything like this when you were recently updating or if you think something else may be causing it since this `remove_outliers` function has remained unchanged. Was the indexing or structure for `TimeSeries.value` returned from `TimeSeries.get()` changed in the new `gwpy`? If you think this *is* a real issue and not just me, then I'm happy to tackle it.

**Comment by duncanmmacleod** [comment]
@macedo22, I'm confused, your traceback seems to call out to this line in `gwdetchar-lasso-correlation`:

```python
    ts[:] = (spline(ts[outliers].times.value) * unit)
```

However, that line does not exist in the repo, what I see is https://github.com/gwdetchar/gwdetchar/blob/8abc6c6ad7bfdc1feab442a3614c6c07411abc7c/bin/gwdetchar-lasso-correlation#L79

Can you confirm that you are running from an unedited copy of that script?

**Comment by macedo22** [comment]
@duncanmmacleod hmm that's strange. I didn't see  until you just pointed it out, but the traceback I included above _was_ from testing it again after making these changes _in response to_ the outlier-removal not working with no changes made to it. So essentially what I mean is Jeff and I _did_ get this traceback before trying to fiddle with it and troubleshoot. In any case, perhaps I missed something originally because I updated again with all the new changes in the last week or so and it works fine, so all is well. My apologies!
